aws-rds-dbcluster/src/main/java/software/amazon/rds/dbcluster/UpdateHandler.java

```diff
@@ -119,4 +139,95 @@ private boolean shouldRemoveFromGlobalCluster(
         return StringUtils.hasValue(previousResourceState.getGlobalClusterIdentifier()) &&
                 StringUtils.isNullOrEmpty(desiredResourceState.getGlobalClusterIdentifier());
     }
+
+    protected ProgressEvent<ResourceModel, CallbackContext> rebootCluster(
+            final AmazonWebServicesClientProxy proxy,
+            final ProxyClient<RdsClient> proxyClient,
+            final ProgressEvent<ResourceModel, CallbackContext> progress,
+            final ResourceModel resourceModel
+    ) {
+        DBCluster cluster;
+        try {
+            cluster = fetchDBCluster(proxyClient, resourceModel);
+        } catch (Exception ex) {
+            return Commons.handleException(progress, ex, DEFAULT_DB_CLUSTER_ERROR_RULE_SET);
+        }
+
+        ProgressEvent<ResourceModel, CallbackContext> result = progress;
+
+        for (final DBClusterMember dbClusterMember: cluster.dbClusterMembers()) {
+            final ProgressEvent<ResourceModel, CallbackContext> p = proxy.initiate("rds::reboot-dbcluster-member-" + dbClusterMember.dbInstanceIdentifier(),
+                            proxyClient,
+                            progress.getResourceModel(),
+                            progress.getCallbackContext())
+                    .translateToServiceRequest(model -> Translator.rebootDbInstanceRequest(dbClusterMember.dbInstanceIdentifier()))
```

Comment:
This bit is super scary tbh. Do we reboot all the instances in the cluster unconditionally?

Reply:
We do not do that unconditionally. `shouldReboot` method is checking for in-place DBInstanceParameterGoup update. For the specific reasons why we want to reboot at all - I am happy to discuss this verbally :)